src/dto/User/create.user.dto.ts

```diff
@@ -0,0 +1,45 @@
+import Joi from 'joi';
+
+const generatePassword = () => {
+    return new RegExp('^[a-zA-Z0-9]{3,30}$')
+};
```

Comment:
Thats wrong name. Also I'm not a big fan of holding it right here. I would prefer file that holds reusable thunks of code like this (bound up with validation). Btw. name should be smth like getPasswordRegExp, another thing is that You could just use syntax like that ([Arrow function - MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions)):
```ts
const generatePassword = () => new RegExp('^[a-zA-Z0-9]{3,30}$')
```
And yet another thing to consider is that if you really need it as function, you could possibly write it as constant named passwordRegExp.